(GH-1973) Add GH issue links to 'Known issues' doc (#1990)
puppetlabs/bolt: documentation/bolt_known_issues.md

# Known issues

## Tasks executed with PowerShell version 2.x or earlier cannot use parameters named `type`

> When executing PowerShell tasks on targets using PowerShell  version 2.x or

Reviewer: ```suggestion
When executing PowerShell tasks on targets using PowerShell version 2.x or
```

> When executing PowerShell tasks on targets using PowerShell version 2.x or
> earlier, you cannot use a task parameter with the name `type`. Since PowerShell
> version 2.x and earlier do not support `type` as a named argument, and
> PowerShell tasks convert parameters to named arguments, a `type` parameter will
> be filtered out before running the task.

Reviewer 2: ```suggestion
When executing PowerShell tasks on targets using PowerShell version 2.x or
earlier, you cannot use a task parameter with the name `type`. Because PowerShell
version 2.x and earlier do not support `type` as a named argument, and
PowerShell tasks convert parameters to named arguments, Bolt will filter out a `type` 
parameter before running the task.
```
Style tip: "Since" is fine grammatically, but we tend to avoid it because it's ambiguous for non native speakers.

Author: Good to know!

  * bcrypt_pbkdf (>= 1.0, < 2.0)
```

A workaround is to use native SSH when you need to authenticate with

> ed25519 keys. When native SSH is enabled, Bolt will use a specified SSH

Reviewer: Does `ed25519` work if users manually install the gems in bolt's environment? That might be easier than using native ssh especially on Windows, where I'm not sure people will just know where ssh executable is installed.

Author: It looks like installing from Bolt's gem environment won't work since it runs into the same issues with building native extensions.

```
C:\Users\Administrator>"C:\Program Files\Puppet Labs\Bolt\bin\gem.bat" install --user-install ed25519
WARNING:  You don't have c:\users\administrator\.gem\ruby\2.5.0\bin in your PATH,
          gem executables will not run.
Building native extensions. This could take a while...
ERROR:  Error installing ed25519:
        ERROR: Failed to build gem native extension.

    current directory: C:/Users/Administrator/.gem/ruby/2.5.0/gems/ed25519-1.2.4/ext/ed25519_ref10
"C:/Program Files/Puppet Labs/Bolt/bin/ruby.exe" -r ./siteconf20200714-2460-14q9lrl.rb extconf.rb
creating Makefile
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.
```

Using native SSH shouldn't be difficult on Windows for any users who have OpenSSH installed, since the default commands Bolt uses (`ssh` and `scp -r`) are both valid.

client to connect to targets instead of the `net-ssh` Ruby library. To
learn more about native SSH, see [native SSH
transport](experimental_features.md#native-ssh-transport).

📖 **Related issues**

- [#1987 - Unable to authenticate with ed25519 keys over SSH transport
  on Windows](https://github.com/puppetlabs/bolt/issues/1987)

## Limited Kerberos support

> Authenticating with Kerberos over SSH is fully supported.

Reviewer: Could probably just retitle the issue "Limited Kerberos support over WinRM" ?
